bentoml/_internal/runner/container.py

```diff
@@ -239,42 +242,101 @@ def payload_to_single(  # pylint: disable=arguments-differ
 
         return pickle.loads(payload.data)
 
-    batch_to_payload = single_to_payload  # type: ignore[assignment]
-    payload_to_batch = payload_to_single
+    @classmethod
+    @inject
+    def batch_to_payloads(  # pylint: disable=arguments-differ
+        cls,
+        batch: "ext.PdDataFrame",
+        indices: t.Sequence[int],
+        batch_dim: int = 0,
+        plasma_db: "ext.PlasmaClient" = Provide[DeploymentContainer.plasma_db],
+    ) -> t.List[Payload]:
+
+        batches = cls.batch_to_batches(batch, indices, batch_dim)
+
+        def to_payload(subbatch: "ext.PdDataFrame"):
+            payload = cls.to_payload(subbatch, plasma_db)
+            payload.meta["batch_size"] = subbatch.shape[batch_dim]
+            return payload
+
+        payloads = [to_payload(subbatch) for subbatch in batches]
+        return payloads
+
+    @classmethod
+    @inject
+    def from_batch_payloads(  # pylint: disable=arguments-differ
+        cls,
+        payloads: t.Sequence[Payload],
+        batch_dim: int = 0,
+        plasma_db: "ext.PlasmaClient" = Provide[DeploymentContainer.plasma_db],
+    ) -> t.Tuple["ext.PdDataFrame", t.List[int]]:
+        batches = [cls.from_payload(payload, plasma_db) for payload in payloads]
+        return cls.batches_to_batch(batches, batch_dim)
 
 
 class DefaultContainer(DataContainer[t.Any, t.List[t.Any]]):
     @classmethod
-    def singles_to_batch(
-        cls, singles: t.Sequence[t.Any], batch_axis: int = 0
-    ) -> t.List[t.Any]:
+    def batches_to_batch(
+        cls, batches: t.Sequence[t.List[t.Any]], batch_dim: int = 0
+    ) -> t.Tuple[t.List[t.Any], t.List[int]]:
         assert (
-            batch_axis == 0
-        ), "Default Runner DataContainer does not support batch_axies other than 0"
-        return list(singles)
+            batch_dim == 0
+        ), "Default Runner DataContainer does not support batch_dim other than 0"
+        batch = []
+        for subbatch in batches:
+            batch.extend(subbatch)
```

Comment:
I made some rough benchmark and it seems that for large amount of list, using `extend` is most efficient way. Any other suggestion?